cocos/renderer/backend/Types.cpp

```diff
@@ -0,0 +1,16 @@
+#include "Types.h"
+
+
+CC_BACKEND_BEGIN
+
+
+bool UniformLocation::operator==(const UniformLocation &other) const
+{
+    return shaderStage == other.shaderStage && location == other.location;
+}
+
+std::size_t UniformLocation::operator()(const UniformLocation &uniform) const {
+    return (((size_t) shaderStage) & 0xFFFFFFF0 )|((size_t)(location << 4));
```

Comment:
`((size_t) shaderStage) & 0xFFFFFFF0` -> `((size_t) shaderStage) | 0xFFFFFFF0` ?

Reply:
Sorry, should be `((size_t) shaderStage) | 0xF`?

Reply:
yes, my fault